### DISCUSSION TITLE
Resue BMS libs for other pojects

### DISCUSSION THREAD

**Discussion by MacGH23** [Louisvdw/dbus-serialbattery#1103] [Ideas]
Hello,
Is it possible to "implement" / description for a resue of the BMS communication for other python projects.
Then the BMS libs, which supports a lot of BMS systems, can be reused more easily.
I'm just developing a automatic DIY Battery charging script where it would be great if would be able to use it for the BMS communication.
Currently it looks like it needs the Victron OS for it.
If it is already possible, please give a hint ;-)
BR

**Comment by MacGH23** [comment]
I modified the project and created a new 
- bmshelper.py. 
Together with the new files 
- uni_bms.by lib (universal BMS)
- uni_bms_test.py  
all the BMS libs can be used by other Python projects.
All other files are unmodified reuse.

Also a standalone test for the BMS without VenusOS is possible.
Would you like to integrate it into your repo or should I create a new one ?

**Reply by mr-manuel** [reply]
Can you share your repo, so that I can take a look?

**Reply by MacGH23** [reply]
update uni_bms to the last updates

**Reply by MacGH23** [reply]
Update to last version

**Comment by MacGH23** [comment]
Sure
I made a quick fork, take a look if you would like to integrate it.
https://github.com/MacGH23/venus-os_dbus-serialbattery
The ini files is not so important, just some features disabled which are not used

**Comment by MacGH23** [comment]
Update to latest version
Any comment about integrating ?

**Reply by mr-manuel** [reply]
I opened a draft PR: https://github.com/mr-manuel/venus-os_dbus-serialbattery/pull/107
